permission denied in dev env when creating assets

Someone reported this to me privately, and I vaguely recall someone else ran into this months ago, please comment if you remember the resolution. 

The person was running Debian, ran `make dev` and the dev env started without the CSS styles loaded. Rerunning `make assets` resulted in this permissions error:

```
$ make assets
docker-compose run --rm web yarn build
WARNING: The AWS_ACCESS_KEY_ID variable is not set. Defaulting to a blank string.
WARNING: The AWS_SECRET_ACCESS_KEY variable is not set. Defaulting to a blank string.
WARNING: The AWS_DEFAULT_REGION variable is not set. Defaulting to a blank string.
WARNING: The S3_BUCKET_NAME variable is not set. Defaulting to a blank string.
WARNING: The APPROVE_REGISTRATIONS variable is not set. Defaulting to a blank string.
Starting openoversight_postgres_1 ... done
yarn run v1.22.4
warning package.json: No license field
$ node-sass --build --recursive OpenOversight/app/static/scss/index.scss -o OpenOversight/app/static/dist
Rendering Complete, saving .css file...
Error: EACCES: permission denied, mkdir '/usr/src/app/OpenOversight/app/static/dist'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Makefile:23: recipe for target 'assets' failed
make: *** [assets] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -16,11 +16,11 @@ create_db: start
 	done
 	@echo "Postgres is up"
 	## Creating database
-	docker-compose run --rm web python ../create_db.py
+	DC_UID=`id -u` docker-compose run --rm web python ../create_db.py
 
 .PHONY: assets
 assets:
-	docker-compose run --rm web yarn build
+	DC_UID=`id -u` docker-compose run --rm web yarn build
 
 .PHONY: dev
 dev: build start create_db populate
@@ -33,13 +33,13 @@ populate: create_db  ## Build and run containers
 	done
 	@echo "Postgres is up"
 	## Populate database with test data
-	docker-compose run --rm web python ../test_data.py -p
+	DC_UID=`id -u` docker-compose run --rm web python ../test_data.py -p
 
 .PHONY: test
 test: start  ## Run tests
 	if [ -z "$(name)" ]; \
-	    then FLASK_ENV=testing docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/; \
-	    else FLASK_ENV=testing docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/ -k $(name); \
+	    then FLASK_ENV=testing DC_UID=`id -u` docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/; \
+	    else FLASK_ENV=testing DC_UID=`id -u` docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/ -k $(name); \
 	fi
 
 .PHONY: cleanassets

--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,14 @@
 default: build start create_db populate test stop clean
 
+export DC_UID := $(shell id -u)
+
 .PHONY: build
 build:  ## Build containers
 	docker-compose build
 
 .PHONY: start
 start: build  ## Run containers
-	DC_UID=`id -u` docker-compose up -d
+	docker-compose up -d
 
 .PHONY: create_db
 create_db: start
@@ -16,11 +18,11 @@ create_db: start
 	done
 	@echo "Postgres is up"
 	## Creating database
-	DC_UID=`id -u` docker-compose run --rm web python ../create_db.py
+	docker-compose run --rm web python ../create_db.py
 
 .PHONY: assets
 assets:
-	DC_UID=`id -u` docker-compose run --rm web yarn build
+	docker-compose run --rm web yarn build
 
 .PHONY: dev
 dev: build start create_db populate
@@ -33,13 +35,13 @@ populate: create_db  ## Build and run containers
 	done
 	@echo "Postgres is up"
 	## Populate database with test data
-	DC_UID=`id -u` docker-compose run --rm web python ../test_data.py -p
+	docker-compose run --rm web python ../test_data.py -p
 
 .PHONY: test
 test: start  ## Run tests
 	if [ -z "$(name)" ]; \
-	    then FLASK_ENV=testing DC_UID=`id -u` docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/; \
-	    else FLASK_ENV=testing DC_UID=`id -u` docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/ -k $(name); \
+	    then FLASK_ENV=testing docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/; \
+	    else FLASK_ENV=testing docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/ -k $(name); \
 	fi
 
 .PHONY: cleanassets

--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ build:  ## Build containers
 
 .PHONY: start
 start: build  ## Run containers
-	docker-compose up -d
+	DC_UID=`id -u` docker-compose up -d
 
 .PHONY: create_db
 create_db: start
@@ -56,7 +56,7 @@ clean: cleanassets stop  ## Remove containers
 
 .PHONY: clean_all
 clean_all: clean stop ## Wipe database
-	rm -rf container_data
+	docker-compose down -v
 
 .PHONY: docs
 docs: ## Build project documentation in live reload for editing

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1,5 +1,9 @@
 version: "3"
 
+volumes:
+  postgres:
+    driver: local
+
 services:
  postgres:
    restart: always
@@ -9,7 +13,7 @@ services:
      POSTGRES_PASSWORD: terriblepassword
      POSTGRES_DB: openoversight-dev
    volumes:
-     - ./container_data/postgres:/var/lib/postgresql/data
+     - postgres:/var/lib/postgresql/data
    ports:
      - "5432:5432"
 
@@ -30,6 +34,7 @@ services:
      FLASK_ENV: "${FLASK_ENV:-development}"
    volumes:
      - ./OpenOversight/:/usr/src/app/OpenOversight/:z
+   user: "${DC_UID}"
    links:
      - postgres:postgres
    expose:

--- a/dockerfiles/web/Dockerfile
+++ b/dockerfiles/web/Dockerfile
@@ -22,11 +22,11 @@ RUN pip3 install --no-cache-dir virtualenv
 RUN virtualenv /ve
 ENV PATH=/ve/bin:$PATH
 
-RUN mkdir /var/www ./node_modules; chown -R www-data /var/www /ve ./node_modules
+RUN mkdir /var/www ./node_modules /.cache /.yarn; chown -R www-data /var/www /ve ./node_modules /.cache /.yarn
 RUN touch /usr/src/app/yarn-error.log
 COPY yarn.lock /usr/src/app/
 RUN chown www-data yarn.lock yarn-error.log
-RUN chmod -R 777 /usr/src/app/
+RUN chmod -R 777 /usr/src/app/ /.cache /.yarn
 
 USER www-data
 

--- a/dockerfiles/web/Dockerfile
+++ b/dockerfiles/web/Dockerfile
@@ -1,10 +1,10 @@
-FROM python:3-stretch
+FROM python:3-buster
 
 WORKDIR /usr/src/app
 
 ENV DEBIAN-FRONTEND noninteractive
 ENV DISPLAY=:1
-ENV GECKODRIVER_VERSION="v0.23.0"
+ENV GECKODRIVER_VERSION="v0.26.0"
 RUN echo "deb http://deb.debian.org/debian stretch-backports main" > /etc/apt/sources.list.d/backports.list
 RUN apt-get update && apt-get install -y xvfb firefox-esr
 RUN apt-get install -y -t stretch-backports libsqlite3-0
@@ -13,22 +13,14 @@ RUN wget https://github.com/mozilla/geckodriver/releases/download/${GECKODRIVER_
 RUN mkdir geckodriver
 RUN tar -xzf geckodriver-${GECKODRIVER_VERSION}-linux64.tar.gz -C geckodriver
 
-RUN apt-get update && apt-get install -y curl
-RUN curl -sL https://deb.nodesource.com/setup_8.x | bash -
-RUN apt-get install -y nodejs
+RUN wget -O - https://deb.nodesource.com/setup_8.x | bash -
+RUN apt-get update && apt-get install -y nodejs npm
 RUN npm install -g yarn
 
-RUN pip3 install --no-cache-dir virtualenv
-RUN virtualenv /ve
-ENV PATH=/ve/bin:$PATH
-
-RUN mkdir /var/www ./node_modules /.cache /.yarn; chown -R www-data /var/www /ve ./node_modules /.cache /.yarn
+RUN mkdir /var/www ./node_modules /.cache /.yarn /.mozilla
 RUN touch /usr/src/app/yarn-error.log
 COPY yarn.lock /usr/src/app/
-RUN chown www-data yarn.lock yarn-error.log
-RUN chmod -R 777 /usr/src/app/ /.cache /.yarn
-
-USER www-data
+RUN chmod -R 777 /usr/src/app/ /var/lib/xkb /.cache /.yarn /.mozilla
 
 COPY requirements.txt dev-requirements.txt /usr/src/app/
 RUN pip3 install --no-cache-dir -r requirements.txt
@@ -36,7 +28,6 @@ RUN pip3 install --no-cache-dir -r requirements.txt
 ARG DOCKER_BUILD_ENV
 RUN test "${DOCKER_BUILD_ENV}" = production || pip3 install --no-cache-dir -r dev-requirements.txt
 
-CMD ["/bin/bash"]
 COPY package.json /usr/src/app/
 RUN yarn
 

